larq_compute_engine/mlir/tf_tfl_passes.cc

```diff
@@ -43,6 +42,9 @@ void AddQuantizationPasses(const mlir::TFL::QuantizationSpecs& quant_specs,
       quant_specs.inference_type != quant_specs.inference_input_type;
   pass_manager->addNestedPass<mlir::FuncOp>(
       mlir::TFL::CreatePostQuantizePass(emit_quant_adaptor_ops));
+  pass_manager->addPass(mlir::TFL::CreateLCEQuantizePass());
+  pass_manager->addNestedPass<mlir::FuncOp>(
+      mlir::TFL::CreatePostQuantizePass(emit_quant_adaptor_ops));
```

Comment:
Stray `tfl.dequantize` ops might prevent the `lq.quantize(tfl.dequantize(x))` -> `lq.quantize(x)` pattern from properly working.

This change runs the pattern again, after unnecessary `tfl.quantize` and `tfl.dequantize` ops have been removed in the `CreatePostQuantizePass`.

before | after
---|---
<img width="272" alt="Screenshot 2022-04-08 at 10 54 16" src="https://user-images.githubusercontent.com/13285808/162413455-982ccba9-04fe-4aeb-aaba-de54a2d9d73c.png"> | <img width="239" alt="Screenshot 2022-04-08 at 11 00 22" src="https://user-images.githubusercontent.com/13285808/162413587-e394e578-7f70-4d2d-a6c4-12edc5b6d274.png">